spec/virtual_total_spec.rb

```diff
@@ -328,4 +345,45 @@ def model_with_children(count)
       end
     end
   end
+
+  describe ".virtual_aggregation with has_many" do
+    let(:author) do
+      Author.create_with_books(1).tap do |author|
+        author.create_books(1, :published => true, :rating => 4)
+        author.create_books(1, :published => true, :rating => 2)
+        author.create_books(1, :published => true)
+      end
+    end
+
+    it "calculates sum locally" do
+      author
+      expect do
+        expect(author.sum_recently_published_books_rating).to eq(6)
```

Comment:
Aside:  This is a really dumb attribute... and I think I wrote it....

I feel that `avg_published_book_rating` would be something a little more realistic.

Reply:
lol. was TOTALLY seeing the same thing.

Once we introduce average (which we don't in our app) - then the `nil` case really starts to show the issue of the `compact`

Reply:
averages would be great - really show the issues with the nils

But then I'd have to come up with a sum test - and I don't want to deal with that right now :(
I'll make this change if you come up with another sum attribute